### DISCUSSION TITLE
Question about compiling process

### DISCUSSION THREAD

**Discussion by naxitoo** [iBotPeaches/Apktool#2939] [Q&A] [answered]
Hi, I've noticed that when adding a new file to the res folder and recompiling, the tool declares in public.xml type, name and also assigns it a unique ID. However, this declaration is not translated to smali (declaring a static field in the R smali class)

Could this be implemented in a near future or is there some sort of drawback? I think it's more practical than having to declare each one manually when it comes to making modifications

**Comment by iBotPeaches** [comment, marked as answer]
I see the benefit to this, but I need to dig into the exact process the generates the R classes to see if thats something we can even feasibly do.

**Reply by naxitoo** [reply]
Oh, I see... Well, thanks for considering it, I hope it is
